docs/AddNewOp.md

```diff
@@ -49,10 +49,11 @@ Once the criteria of proposing new operator/function has been satisfied, you wil
     1. The testing examples will be extracted to the doc.
     2. We also generate binary data for it.
     3. Example: https://github.com/onnx/onnx/blob/master/onnx/backend/test/case/node/abs.py
-5. Update the documentation and generate the test data.
+5. Add at least one automatic upgrade test for your operator in https://github.com/onnx/onnx/blob/master/onnx/test/automatic_upgrade_test.py
```

Comment:
Is this step required for new operator?
In case of op updates, what happens if automatic upgrade is not possible?

Reply:
I see some details are present in Versioning.md... can you add them here as well as people generally use this doc to understand all the steps involved

Reply:
> Is this step required for new operator?

Yes because the new tests check that all operators are tested at least once (`test_ops_tested`)

> In case of op updates, what happens if automatic upgrade is not possible?

What do you mean by this? As long as the old op's functionality is a subset of the new one's, upgrade is always possible. Maybe the only case in which this got violated is `batch_normalization_13_14.h` which I added in this PR (opset 14 disabled outputs 4 and 5). Currently the automatic upgrade tests are not testing this upgrade path (they are FAR from testing all the possible upgrade paths from all possible settings for all ops anyways). In the future we can extend these new tests to also add upgrade paths which are expected to fail at some step.

> Can you add more clarification in the doc
> I see some details are present in Versioning.md... can you add them here as well as people generally use this doc to understand all the steps involved

Versioning.md says to add an adapter whenever a change is made (and don't add a test because the automatic tests will test the adapter), while this is about adding a test whenever a new op is introduced (and don't add an adapter because there's nothing to adapt), so I don't see much overlap. But I have added more information about the basic idea of the new tests.

Reply:
thanks for the clarification.
How are we enforcing\gating PRs on these tests... Will these tests run as part of python test suite and fail if someone did not add a test for new op?

Reply:
> thanks for the clarification.
> How are we enforcing\gating PRs on these tests... Will these tests run as part of python test suite and fail if someone did not add a test for new op?

Yes. The CI will fail in these 2 cases:
1) A new op is introduced but an automatic upgrade test is not added
2) An op is changed but an upgrade adapter is not added